language/move-prover/boogie-backend-v2/src/boogie_wrapper.rs

```diff
@@ -612,21 +609,15 @@ pub enum ValueArrayRep {
 #[derive(Debug)]
 pub struct Model {
     vars: BTreeMap<ModelValue, ModelValue>,
-    tracked_exps: BTreeMap<ExpDescriptor, Vec<ModelValue>>,
-    value_array_rep: ValueArrayRep,
+    vector_theory: VectorTheory,
 }
 
 impl Model {
     /// Create a new model.
     fn new(wrapper: &BoogieWrapper<'_>) -> Self {
         Model {
             vars: Default::default(),
-            tracked_exps: Default::default(),
-            value_array_rep: if wrapper.options.vector_using_sequences {
-                ValueArrayRep::ValueArrayIsSeq
-            } else {
-                ValueArrayRep::ValueArrayIsMap
-            },
+            vector_theory: wrapper.options.vector_theory,
```

Comment:
General question:  Are you supporting model extraction and visualization for all vector representations?

Reply:
Only BoogieArray and SmtSeq for now.